test/publisher.ts

```diff
@@ -111,49 +115,57 @@ describe('Publisher', () => {
       (global as any).setTimeout = util.noop;
     });
 
+    afterEach(() => sandbox.restore());
+
     after(() => {
-      global.setTimeout = globalSetTimeout;
+      // tslint:disable-next-line no-any
+      (global as any).setTimeout = globalSetTimeout;
     });
 
-    it('should throw an error when data is not a buffer', () => {
-      assert.throws(() => {
-        publisher.publish('hello', {}, assert.ifError);
-      }, /Data must be in the form of a Buffer\./);
-    });
+    // it('should throw an error when data is not a buffer', () => {
```

Comment:
need advice, as we explicitly declaring types, some of test cases may not required to check required parameters or parameters type!